### ISSUE TITLE
Unable to install the module on a new installation / Area code is not set

### ISSUE THREAD

**Issue by pierre-jeanbonazzi** [Adyen/adyen-magento2#2540]
**Describe the bug**
Hello, 

I ran into a problem when installing the adyen/module-payment module in version 9.2.0 for a Magento 2.4.6-p4 for the first time.

In fact, after making the command `composer require adyen/module-payment`, Magento asks me to make a `setup:upgrade`. 

![image](https://github.com/Adyen/adyen-magento2/assets/162571289/22210684-e9be-4a42-8f27-0eed1224e05f)

However, when running setup:upgrade. I get an error "Area code is not set". 

![image](https://github.com/Adyen/adyen-magento2/assets/162571289/5a23ee42-3b8e-4de0-9d84-4f675db500d8)

**To Reproduce**
- Installing the module for the first time with `composer require adyen/module-payment`
- Run `bin/magento setup:upgrade`

**Magento version**
Magento Commerce 2.4.6-p4

**Plugin version**
Adyen_Payment 9.2.0

**Comment by candemiralp** [comment]
Hello @pierre-jeanbonazzi,

Thank you for creating this issue. We are currently investigating this issue and will inform you soon.

Best Regards,
Can

**Comment by candemiralp** [comment]
Hello @pierre-jeanbonazzi,

We are not able to reproduce the issue on a fresh instance with the following commands.

```
composer require adyen/module-payment
bin/magento setup:upgrade
```

After following commands, plugin has been installed correctly. Do you have any other insight to help us to reproduce the issue?

Best Regards,
Can

**Comment by pierre-jeanbonazzi** [comment]
Hello @candemiralp , 

I'm sorry, but unfortunately I've only done the steps described in my initial message.

After taking a closer look and modifying the files : if I delete the file `Setup/Recurring.php` and the file `Test/Unit/Setup/RecurringTest.php`, the installation goes ahead without any problems.

I found this while searching for your old issues and came across this one: https://github.com/Adyen/adyen-magento2/pull/2509

I made the change based on your fix

Best regards,

**Comment by candemiralp** [comment]
Hello @pierre-jeanbonazzi,

Thanks for the feedback. Could you please re-try the whole process with the develop branch?

`composer require adyen/module-payment:dev-develop`

Best Regards,
Can

**Comment by pierre-jeanbonazzi** [comment]
Hello @candemiralp,

I've just tried it with the develop branch: the "Area code is not set" error is still there when I do a `bin/magento setup:upgrade` but it appears in a different place in the process.
![image](https://github.com/Adyen/adyen-magento2/assets/162571289/85fa9a94-6b98-4e4a-ba2a-e5b22e472262)

On the other hand, I no longer get the error on Magento and I can access my environment without any problem. I can get the module to work and make payments.

Best regards,

**Comment by pierre-jeanbonazzi** [comment]
Hello @candemiralp ,

Without bothering you, I'd like to ask you again about the problem I encountered with your module. 

Do you have any solutions for this ?

Thank you in advance, 

Best regards,

**Comment by pierre-jeanbonazzi** [comment]
Issue resolved with [9.5.1](https://github.com/Adyen/adyen-magento2/tree/9.5.1) version
